programs/adapter-tulip/src/lib.rs

```diff
@@ -129,100 +104,228 @@ pub mod adapter_tulip {
-            AccountMeta::new(ctx.remaining_accounts[4].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[5].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[6].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[7].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[8].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[9].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[10].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[11].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[12].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[13].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[14].key(), false),
-            AccountMeta::new_readonly(ctx.remaining_accounts[15].key(), false),
-            AccountMeta::new_readonly(ctx.remaining_accounts[16].key(), false),
-            AccountMeta::new_readonly(ctx.remaining_accounts[17].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[18].key(), false),
-            AccountMeta::new(ctx.remaining_accounts[19].key(), false),
-        ];
-
-        let mut withdraw_raydium_data = vec![];
-        withdraw_raydium_data.append(&mut sighash_arr.try_to_vec()?);
-        withdraw_raydium_data.append(&mut input_struct.share_amount.try_to_vec()?);
-
-        let withdraw_raydium_ix = Instruction {
-            program_id: ctx.accounts.base_program_id.key(),
-            accounts: withdraw_raydium_accounts,
-            data: withdraw_raydium_data,
+        let (lp_or_token_a_amount, token_b_amount) = match input_struct.farm_type_0 {
+            0 => {
+                let mut token_account_and_balance =
+                    load_token_account_and_balance(ctx.remaining_accounts, 13);
+
+                // reference: https://github.com/sol-farm/tulipv2-sdk/blob/main/vaults/src/instructions/deposit_tracking.rs#L38
+                let mut withdraw_deposit_tracking_data =
+                    sighash("global", "withdraw_deposit_tracking").try_to_vec()?;
+                withdraw_deposit_tracking_data.append(&mut input_struct.share_amount.try_to_vec()?);
+                withdraw_deposit_tracking_data.append(&mut input_struct.farm_type_0.try_to_vec()?);
+                withdraw_deposit_tracking_data.append(&mut input_struct.farm_type_1.try_to_vec()?);
+
+                let accounts = load_remaining_accounts(
+                    ctx.remaining_accounts,
+                    vec![0, 15, 16, 20, 21, 22, 12, 14, 1],
+                );
+
+                let withdraw_deposit_tracking_ix = Instruction {
+                    program_id: ctx.accounts.base_program_id.key(),
+                    accounts,
+                    data: withdraw_deposit_tracking_data,
+                };
+                invoke(
+                    &withdraw_deposit_tracking_ix,
+                    &[
```

Comment:
Can replace these with accounts derived from `load_remaining_account`?

Reply:
`load_remaining_accounts` return `Vec<AccountMeta>` which will used in constructing `Instruction`, but during invocation it require `[AccountInfo]`, that's why I didn't use it.

Reply:
I've tried to convert `Vec<AccountMeta>` back to `[AccountInfo]` but it didn't work since rust need static size of array.

Reply:
Gotcha. How about implementing a helper function here to convert `Vec<AccountMeta>` to `[AccountInfo]`? In this case the hand-made wrapping can be entirely removed.

Reply:
https://stackoverflow.com/questions/29570607/is-there-a-good-way-to-convert-a-vect-to-an-array

Is this helpful?

Reply:
I'd tried this before but type `[AccountInfo]` is different from `[AccountInfo; n]`, so this doesn't work.

Reply:
Have you tried `slice`?

Example: https://gist.github.com/rust-play/2014447301c7b42f66dbbe22dda4963d